base_location_nuts/wizard/nuts_import.py

```diff
@@ -132,16 +132,16 @@ def _download_nuts(self, url_base=None, url_path=None, url_params=None):
         logger.info('Download successfully %d bytes' %
                     len(res_request.content))
         # Workaround XML: Remove all characters before <?xml
-        pattern = re.compile(r'^.*<\?xml'.encode('utf-8'), re.DOTALL)
+        pattern = re.compile(rb'^.*<\?xml', re.DOTALL)
         content_fixed = re.sub(pattern, b'<?xml',
                                res_request.content)
-        if not re.match(r'<\?xml'.encode('utf-8'), content_fixed):
+        if not re.match(rb'<\?xml', content_fixed):
             raise UserError(_('Downloaded file is not a valid XML file'))
         return content_fixed
 
     @api.model
     def _load_countries(self):
-        for k in list(self._countries.keys()):
+        for k in self._countries.keys():
```

Comment:
Please remove `.keys()`

Reply:
It's not finished yet... I have to take a look also to re.compile thing.

Reply:
Haha OK, too quick review 😋